src/main.cpp

```diff
@@ -160,12 +160,15 @@ class BroadcastMessageSenderTask : public AbstractTask<3 * configMINIMAL_STACK_S
                                     BspContainer::getBSP(), m_logger);
 
         while (NetworkContainer::getNetworkManager().getNetworkStatus() !=
-                   NetworkStatus::Connected &&
+                   NetworkStatus::Connected ||
                BspContainer::getBSP().getHiveMindUUID() == 0) {
-            Task::delay(500);
+            Task::delay(100);
```

Comment:
You could have one task that only starts the others once the connection is established and the id is obtained. Though it's probably overkill and not really necessary, just that I saw you have the same logic in all your task. I'll let you decide if it's necessary or not.

Reply:
I'll probably refac the main at some point, just not in this PR